presto-thrift-connector/src/test/java/com/facebook/presto/connector/thrift/util/TestRetryDriver.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.connector.thrift.util;
+
+import com.google.common.util.concurrent.ListenableFuture;
+import com.google.common.util.concurrent.ListeningExecutorService;
+import com.google.common.util.concurrent.MoreExecutors;
+import org.testng.annotations.Test;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Executors;
+
+import static org.testng.Assert.fail;
+
+public class TestRetryDriver
+{
+    private static class FutureGenerator
+            implements Callable<ListenableFuture<Integer>>
+    {
+        private final int successAttempt;
+        private int currentAttempt;
+        private ListeningExecutorService executor;
```

Comment:
typically you want to move executor to an outer scope. Also, you need to shut it down at some point.